raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -3279,7 +3279,7 @@ private long pollUnattachedAsVoter(UnattachedState state, long currentTimeMs) {
             // If shutting down, then remain in this state until either the
             // shutdown completes or an epoch bump forces another state transition
             return shutdown.remainingTimeMs();
-        } else if (state.hasElectionTimeoutExpired(currentTimeMs)) {
+        } else if (state.hasFetchTimeoutExpired(currentTimeMs)) {
```

Comment:
If I understand correctly, we should use fetch timeout instead of election timeout when aligning with the follower's transition to the prospective state.